hazelcast/src/test/java/com/hazelcast/cache/impl/CacheCreateUseDestroyTest.java

```diff
@@ -0,0 +1,223 @@
+import javax.cache.event.CacheEntryListenerException;
+import javax.cache.spi.CachingProvider;
+import java.io.Closeable;
+import java.io.IOException;
+import java.io.Serializable;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+
+import static com.hazelcast.config.EvictionConfig.MaxSizePolicy.USED_NATIVE_MEMORY_PERCENTAGE;
+import static com.hazelcast.config.InMemoryFormat.BINARY;
+import static com.hazelcast.config.InMemoryFormat.NATIVE;
+import static com.hazelcast.config.InMemoryFormat.OBJECT;
+import static java.lang.String.format;
+import static org.hamcrest.Matchers.not;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+@RunWith(Parameterized.class)
+@UseParametersRunnerFactory(HazelcastParametersRunnerFactory.class)
+@Category({QuickTest.class})
+public class CacheCreateUseDestroyTest extends HazelcastTestSupport {
+
+    public static final MemorySize NATIVE_MEMORY_SIZE = new MemorySize(32, MemoryUnit.MEGABYTES);
+
+    @Parameters(name = "{0}")
+    public static Collection parameters() {
+        return Arrays.asList(
+                new Object[]{OBJECT},
+                new Object[]{BINARY},
+                new Object[]{NATIVE}
+        );
+    }
+
+    @Parameter
+    public InMemoryFormat inMemoryFormat;
+
+    protected CacheManager defaultCacheManager;
+    protected Cache<String, String> cache;
+    protected ICacheService cacheService;
+
+    @Before
+    public void setup() {
+        assumptions();
+        TestHazelcastInstanceFactory factory = createHazelcastInstanceFactory(1);
+        HazelcastInstance member = factory.newHazelcastInstance(getConfig());
+        CachingProvider provider = Caching.getCachingProvider();
+        defaultCacheManager = provider.getCacheManager(null, null,
+                HazelcastCachingProvider.propertiesByInstanceItself(member));
+        cacheService = getNode(member).getNodeEngine().getService(ICacheService.SERVICE_NAME);
+        CacheEntryListenerFactory.listener = null;
+    }
+
+    protected void assumptions() {
+        Assume.assumeThat(inMemoryFormat, not(NATIVE));
+    }
+
+    @Override
+    protected Config getConfig() {
+        Config config = super.getConfig();
+        CacheSimpleEntryListenerConfig entryListenerConfig = new CacheSimpleEntryListenerConfig();
+        entryListenerConfig.setCacheEntryListenerFactory("com.hazelcast.cache.impl.CacheCreateUseDestroyTest$CacheEntryListenerFactory");
+        entryListenerConfig.setOldValueRequired(true);
+        entryListenerConfig.setSynchronous(true);
+        CacheSimpleConfig cacheSimpleConfig = new CacheSimpleConfig()
+                .setName("cache*")
+                .setInMemoryFormat(inMemoryFormat)
+                .setStatisticsEnabled(true)
+                .setManagementEnabled(true)
+                .setCacheEntryListeners(Collections.singletonList(entryListenerConfig));
+        if (inMemoryFormat == NATIVE) {
```

Comment:
With `Assume.assumeThat(inMemoryFormat, not(NATIVE));` in the `@Setup` method this should never called, should it?